python/pyarrow/tests/test_convert_builtin.py

```diff
@@ -863,6 +867,34 @@ def test_datetime_subclassing():
     assert arr_us[0].as_py() == datetime.datetime(2007, 7, 13, 1,
                                                   23, 34, 123456)
 
+    data = [
+        MyTimedelta(1, 1, 1000),
```

Comment:
Is it possible to test the conversion with more interesting values?

Reply:
What values would you find interesting? (thinking about specific corner cases?)

(it's mainly the fact that the subclass is working that is being tested here)

Reply:
Something less trivial than days and seconds having the same value.